Core/include/Input.h

```diff
@@ -1,87 +1,82 @@
-#pragma once
-#include <unordered_map>
-#include <glad/glad.h> // Include glad first
+#include <queue>
+#include <glad/glad.h>
 #include <GLFW/glfw3.h>
 #include <iostream>
+#include <mutex>
 
+#include "include/Body.h"
 
 class Input
 {
 public:
-    // static maps that keep track of the current state of keypresses & mouse clicks (pressed or unpress)
-    static std::unordered_map<int, bool> keys;
-    static std::unordered_map<int, bool> mouseButtons;
+    enum ActionType { PRESS, RELEASE, HOLD };
 
+    struct InputEvent {
+        int code;
```

Comment:
You call it code here but you assign it a key, is there a reason you call it code?

Reply:
it's keycode basically
sorrt for the confusion